handlers.go

```diff
@@ -54,23 +54,29 @@ func receiveHandler(producer *kafka.Producer, serializer Serializer) func(c *gin
 			return
 		}
 
-		metrics, err := processWriteRequest(&req)
+		metricsPerTopic, err := processWriteRequest(&req)
 		if err != nil {
 			c.AbortWithStatus(http.StatusInternalServerError)
 			logrus.WithError(err).Error("couldn't process write request")
 			return
 		}
 
-		for _, metric := range metrics {
-			err := producer.Produce(&kafka.Message{
-				TopicPartition: kafkaPartition,
-				Value:          metric,
-			}, nil)
+		for topic, metrics := range metricsPerTopic {
+			for _, metric := range metrics {
+				t := &topic
```

Comment:
I would suggest to move this assignment to the previous loop, because `topic` doesn't change on each `metric` iteration.

```suggestion
```

Reply:
yep, moved the partition (at all) there